File: 941-sort-array-by-parity/sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        forward = 0
        after = len(nums) - 1
        while forward < after :
            if nums[forward] % 2 == 1 and nums[after] % 2 == 0:
                nums[forward], nums[after] = nums[after], nums[forward]
            if nums[forward] % 2 == 0:
                forward += 1
            if nums[after] % 2 == 1:
                after -= 1
        return nums
        

File: 941-sort-array-by-parity/sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int forward = 0;
        int after = nums.size() - 1;
        while (forward < after) {
            if (nums[forward] % 2 == 1 && nums[after] % 2 == 0) {
                int temp = nums[forward];
                nums[forward] = nums[after];
                nums[after] = temp;
                forward++;
                after--;
            }
            if (nums[forward] % 2 == 0 ) {
                forward++;
            }
            if (nums[after] % 2 == 1) {
                after--;
            }
        }
        return nums;
    }
};